jekyll/_cci2/bundling-config.md

```diff
@@ -0,0 +1,905 @@
+- `environment`
+- `working_directory`
+- `shell`
+- `resource_class`
+
+A simple example of using an executor:
+
+```yaml
+version: 2.1
+executors:
+  my-executor:
+    docker:
+      - image: circleci/ruby:volatile
+
+jobs:
+  my-job:
+    executor: my-executor
+    steps:
+      - run: echo outside the executor
+```
+
+In the above example the executor `my-executor` is passed as the single value of the key `executor`. Alternatively, you can pass `my-executor` as the value of a `name` key under `executor` -- this method is primarily employed when passing parameters to executor invocations (see below):
+
+```yaml
+jobs:
+  my-job:
+    executor:
+      name: my-executor
+    steps:
+      - run: echo outside the executor
+```
+
+### Common uses of executors
+{:.no_toc}
+
+Executors in configuration were designed to enable re-use of a defined execution environment in multiple jobs in the `config.yml` file.
+
+<!---
+2. Allowing an orb to define the executor used by all of its commands. This allows users to execute the commands of that orb in the execution environment defined by the orb's author.
+-->
+
+#### Example of Using an Executor Declared in config.yml in Many Jobs
+{:.no_toc}
+
+Imagine you have several jobs that you need to run in the same Docker image and working directory with a common set of environment variables. Each job has distinct steps, but should run in the same environment. You have three options to accomplish this:
+
+1. Declare each job with repetitive `docker` and `working_directory` keys.
+2. Use YAML anchors to achieve some reuse.
+3. Declare an executor with the values you want, and invoke it in your jobs.
+
+With an executor declaration your configuration might look something like: 
+
+**Without executors**
+```yaml
+jobs:
+  build:
+    docker:
+      - image: clojure:lein-2.8.1
+    working_directory: ~/project
+    environment:
+      MYSPECIALVAR: "my-special-value"
+      MYOTHERVAR: "my-other-value"
+    steps:
+      - checkout
+      - run: echo "your build script would go here"
+      
+  test:
+    docker:
+      - image: clojure:lein-2.8.1
+    working_directory: ~/project
+    environment:
+      MYSPECIALVAR: "my-special-value"
+      MYOTHERVAR: "my-other-value"
+    steps:
+      - checkout
+      - run: echo "your test commands would run here"
+```
+    
+**Same Code, With executors**
+```yaml
+executors:
+  lein_exec:
+    docker:
+      - image: clojure:lein-2.8.1
+    working_directory: ~/project
+    environment:
+      MYSPECIALVAR: "my-special-value"
+      MYOTHERVAR: "my-other-value"
+    
+jobs:
+  build:
+    executor: lein_exec
+    steps:
+      - checkout
+      - run: echo "hello world"
+      
+  test:
+    executor: lein_exec
+    environment:
+      TESTS: unit
+    steps:
+      - checkout
+      - run: echo "how are ya?"
+```
+
+You can also refer to executors from other orbs. Users of an orb can invoke its executors. For example, `foo-orb` could define the `bar` executor:
```

Comment:
"orbs"

Reply:
(this applies to the next few snippets too -- probably doesn't belong here)

Reply:
thanks, commented these out